Wikipedia/Code/WMFProxyServer.m

```diff
@@ -189,136 +189,122 @@ - (NSURL*)proxyURLForImageURLString:(NSString*)imageURLString {
-
-        NSString* after = [regex replacementStringForResult:result
-                                                   inString:mutableString
-                                                     offset:offset
-                                                   template:@"$4"];
-
-        NSString* replacement = [NSString stringWithFormat:@"%@%@\"%@\"%@",
-                                 before,
-                                 srcsetKey,
-                                 [self stringByReplacingURLsWithProxyURLsInSrcsetValue:srcsetValue],
-                                 after
-                                ];
-
-        [mutableString replaceCharactersInRange:resultRange withString:replacement];
-
-        offset += [replacement length] - resultRange.length;
+        resizedSrc = [NSString pathWithComponents:srcPathComponents];
+        if (![resizedSrc hasPrefix:@"//"] && [resizedSrc hasPrefix:@"/"]) {
+            resizedSrc = [@[@"/", resizedSrc] componentsJoinedByString:@""];
+        }
+        src = resizedSrc;
     }
-    return mutableString;
-}
-
-- (NSString*)stringByReplacingURLsWithProxyURLsInSrcsetValue:(NSString*)srcsetValue {
-    NSAssert(![srcsetValue containsString:@"<"] && ![srcsetValue containsString:@">"], @"This method should only operate on an html img tag's 'srcset' value substring - not entire image tags.");
-
-    NSArray* pairs         = [srcsetValue componentsSeparatedByString:@","];
-    NSMutableArray* output = [[NSMutableArray alloc] init];
-    for (NSString* pair in pairs) {
-        NSString* trimmedPair = [pair stringByTrimmingCharactersInSet:[NSCharacterSet whitespaceCharacterSet]];
-        NSArray* parts        = [trimmedPair componentsSeparatedByCharactersInSet:[NSCharacterSet whitespaceCharacterSet]];
-        if (parts.count == 2) {
-            NSString* url     = parts[0];
-            NSString* density = parts[1];
-            [output addObject:[NSString stringWithFormat:@"%@ %@", [self proxyURLForImageURLString:url].absoluteString, density]];
-        } else {
-            [output addObject:pair];
+    
+    if (src) {
+        NSString *srcWithProxy = [self proxyURLForImageURLString:src].absoluteString;
+        if (srcWithProxy) {
+            NSString *newSrcAttribute = [@[@"src=\"", srcWithProxy, @"\""] componentsJoinedByString:@""];
+            [newImageTagContents replaceCharactersInRange:srcAttributeRange withString:newSrcAttribute];
         }
     }
-    return [output componentsJoinedByString:@", "];
+    
+    
+    [newImageTagContents replaceOccurrencesOfString:@"srcset" withString:@"data-srcset-disabled" options:0 range:NSMakeRange(0, newImageTagContents.length)]; //disable the srcset since we put the correct resolution image in the src
+    
+    if (resizedSrc) {
+        [newImageTagContents appendString:@" data-image-resized=\"true\""]; //the javascript looks for this so it doesn't try to change the src again
```

Comment:
For future reference, I added `style="border-radius: 60px;"` to this string for testing. I also enabled `enableDebugBorders` in widenImages.js. That way you could easily see which native and js transforms took effect. Looked good.